File: test_repo/sample.cpp
#include <iostream>
#include <string>

class Player {
public:
    float health;
    float max_health;
    float x;
    float y;
    float speed;

    Player() : health(100.0f), max_health(100.0f), x(0.0f), y(0.0f), speed(5.0f) {}

    void calculate_player_health(float damage) {
        health -= damage;
        if (health < 0) health = 0;
        std::cout << "C++ Player took " << damage << " damage. Health: " << health << std::endl;
    }

    void move(float dx, float dy) {
        x += dx * speed;
        y += dy * speed;
        std::cout << "C++ Player moved to (" << x << ", " << y << ")" << std::endl;
    }
};

int main() {
    Player player;
    player.calculate_player_health(25.0f);
    player.move(1.0f, 1.0f);
    return 0;
}

File: test_repo/sample.go
package main

import "fmt"

type Player struct {
	Health    float64
	MaxHealth float64
	X         float64
	Y         float64
	Speed     float64
}

func (p *Player) calculate_player_health(damage float64) {
	p.Health -= damage
	if p.Health < 0 {
		p.Health = 0
	}
	fmt.Printf("Go Player took %v damage. Health: %v\n", damage, p.Health)
}

func (p *Player) move(dx, dy float64) {
	p.X += dx * p.Speed
	p.Y += dy * p.Speed
	fmt.Printf("Go Player moved to (%v, %v)\n", p.X, p.Y)
}

func main() {
	player := Player{Health: 100, MaxHealth: 100, Speed: 5}
	player.calculate_player_health(20)
	player.move(1, 0)
}
